internal/config/from_env.go

```diff
@@ -57,15 +55,28 @@ func (c *envConfig) GetWithSource(hostname, key string) (string, string, error)
 
 func (c *envConfig) CheckWriteable(hostname, key string) error {
 	if hostname != "" && key == "oauth_token" {
-		envName := GITHUB_TOKEN
-		if ghinstance.IsEnterprise(hostname) {
-			envName = GITHUB_ENTERPRISE_TOKEN
+		if _, env, found := AuthTokenFromEnv(hostname); found {
+			return fmt.Errorf("read-only token in %s cannot be modified", env)
 		}
+	}
+
+	return c.Config.CheckWriteable(hostname, key)
+}
 
-		if os.Getenv(envName) != "" {
-			return fmt.Errorf("read-only token in %s cannot be modified", envName)
+func AuthTokenFromEnv(hostname string) (string, string, bool) {
+	if ghinstance.IsEnterprise(hostname) {
+		if token, found := os.LookupEnv(GH_ENTERPRISE_TOKEN); found {
```

Comment:
`LookupEnv` will return a positive `found` if the environment variable was set but empty. I would suggest that we avoid this and to treat empty values the same way as if the environment variable was unset.

E.g. `GH_TOKEN="" GITHUB_TOKEN="hello"` 👉 GITHUB_TOKEN should get used, since GH_TOKEN is unusable.

Or did you intend for `GH_TOKEN=""` to be able to "blank out" a GITHUB_TOKEN and ensure that the config file gets used? There is merit to that feature, but in that case we should document it (either in man pages or the code) and provide test coverage.

Reply:
I did change the behavior here on purpose. I was thinking that because environment variables need to be purposefully set, it seems unlikely that they would be set to empty strings unless that was what the user intended and we should respect that intention. I don't have strong feelings though and am comfortable with changing the logic back to where an empty string is equivalent to being unset. If we go that route I will remove the `found` return value as well since it doesn't make sense in that case. 

In the current implementation, I don't think `GH_TOKEN=""` will actually result in "blanking out" the `GITHUB_TOKEN` and forcing the config file to be read because line 49 will short circuit and return the empty string, but it would be a simple change to add in that feature. I think adding in this feature actually makes it more confusing to understand the order of precedence of these auth tokens though, so I would be okay with not adding it now and waiting to see if that is something our users want.

Reply:
I don't know how likely is it that people set any of these environment variables to blank values, but we've treated `GITHUB_TOKEN=""` (empty value) the same as the environment variable not being set at all, and without a strong reason to change this, I would rather suggest keeping backwards compatibility.

This PR is about adding a new feature (GH_TOKEN support), but it looks like it currently also changes the behavior of how `GITHUB_TOKEN=""` is treated.